src/edu/csus/ecs/pc2/clics/API202306/CLICSClarification.java

```diff
@@ -0,0 +1,246 @@
+                    ElementId [] destGroups = clarAns.getAllDestinationsGroup();
+
+                    // first let's look at destination teams, if any
+                    if(destTeams != null && destTeams.length > 0){
+                        // Use first team in the list for backward compatibilty of only supporting a single dest team
+                        // CLICS does not (yet) allow for more than one
+                        to_team_id = "" + destTeams[0].getClientNumber();
+                        // Implement PC2 extension of proposed CLICS change - array of teams.
+                        ArrayList<String> teamIdList = new ArrayList<String>();
+                        for(ClientId destClient : destTeams) {
+                            teamIdList.add("" + destClient.getClientNumber());
+                        }
+                        to_team_ids = teamIdList.toArray(new String[teamIdList.size()]);
+                    }
+                    // next, let's look at destination groups, if any
+                    // it should be noted that if to_team_id is null at this point, it will remain null, implying
+                    // the response to the clar was sent to all teams.  It's a reasonable compromise since the 2023-06 spec
+                    // does not have a notion of group replies.
+                    if(destGroups != null && destGroups.length > 0) {
+                        ArrayList<String> groupIdList = new ArrayList<String>();
+
+                        for(ElementId groupEle : destGroups) {
+                            Group group = model.getGroup(groupEle);
+                            if(group != null) {
+                                groupIdList.add("" + group.getGroupId());
+                            }
+                            to_group_ids = groupIdList.toArray(new String[groupIdList.size()]);
+                        }
+                    }
+                }
+            }
+            // Announcements do not have a question.
+            if(clar.isAnnounced()) {
+                reply_to_id = null;
+            } else {
+                reply_to_id = clar.getElementId().toString();
+            }
+            text = clarAns.getAnswer();
+            time = Utilities.getIso8601formatterWithMS().format(clarAns.getDate());
+            contest_time = ContestTime.formatTimeMS(clarAns.getElapsedMS());
+        } else {
+            // the request goes to a judge not a team, so to_team_id and reply_to_id is null
+            // fill in question and time fields
+            text = clar.getQuestion();
+            time = Utilities.getIso8601formatterWithMS().format(clar.getCreateDate());
+            contest_time = ContestTime.formatTimeMS(clar.getElapsedMS());
+        }
+        // if not a general clar and it's not a special category clar, then we need to supply the problem id.
+        if (!clar.getProblemId().equals(model.getGeneralProblem()) && model.getCategory(clar.getProblemId()) == null) {
```

Comment:
`clar.getProblemId()` returns an `ElementId` (defined in `Submission`, the parent class of `Clarification`).  OTOH, `model.getGeneralProblem()` returns a `Problem` object.  It doesn't make sense to invoke `.equals()` to compare an `ElementId` with a `Problem`; this will always return `false`.

Reply:
Yes, that's a bug.  Fixed in next push.